nixos/tests/adguardhome.nix

```
@@ -2,41 +2,81 @@
   name = "adguardhome";
 
   nodes = {
-    nullConf = { ... }: { services.adguardhome = { enable = true; }; };
+    nullConf = { services.adguardhome.enable = true; };
 
-    emptyConf = { lib, ... }: {
+    emptyConf = {
       services.adguardhome = {
         enable = true;
+
+        settings = { };
+      };
+    };
+
+    schemaVersionBefore23NullConf = { pkgs, ... }: {
+      services.adguardhome = {
+        enable = true;
+
+        package = pkgs.adguardhome.overrideAttrs (_: rec {
+          version = "0.107.32";
+
+          src = {
+              x86_64-darwin = pkgs.fetchzip {
+                sha256 = "sha256-DM/EhbKp2sM5OnvFrb2GcjhgMN+9kuVW33wHVSxcoLo=";
+                url =
+                  "https://github.com/AdguardTeam/AdGuardHome/releases/download/v0.107.32/AdGuardHome_darwin_amd64.zip";
+              };
+              aarch64-darwin = pkgs.fetchzip {
+                sha256 = "sha256-rzGFcWou9+rDJuxqY6utnT54jAVLG2oj3Og8Fr8pibU=";
+                url =
+                  "https://github.com/AdguardTeam/AdGuardHome/releases/download/v0.107.32/AdGuardHome_darwin_arm64.zip";
+              };
+              i686-linux = pkgs.fetchurl {
+                sha256 = "sha256-JkDQ9kmXuwcFXcfyDYwgI/NHqJYZj/PbhZfqlL0jvnw=";
+                url =
+                  "https://github.com/AdguardTeam/AdGuardHome/releases/download/v0.107.32/AdGuardHome_linux_386.tar.gz";
+              };
+              x86_64-linux = pkgs.fetchurl {
+                sha256 = "sha256-YBE4yL63Ee1UmkE7pIbLKOLLhiY5DpN1t8i/ovQOKXo=";
+                url =
+                  "https://github.com/AdguardTeam/AdGuardHome/releases/download/v0.107.32/AdGuardHome_linux_amd64.tar.gz";
+              };
+              aarch64-linux = pkgs.fetchurl {
+                sha256 = "sha256-Lj4eCcMgpy2AlGHKA/xRxZc2HV3llykWwVeTUlHXvyI=";
+                url =
+                  "https://github.com/AdguardTeam/AdGuardHome/releases/download/v0.107.32/AdGuardHome_linux_arm64.tar.gz";
+              };
+              armv6l-linux = pkgs.fetchurl {
+                sha256 = "sha256-vAP/xc3IYMCtDbog2qB3kK5ftbsrk8meLD0IGx9tNa8=";
+                url =
+                  "https://github.com/AdguardTeam/AdGuardHome/releases/download/v0.107.32/AdGuardHome_linux_armv6.tar.gz";
+              };
+              armv7l-linux = pkgs.fetchurl {
+                sha256 = "sha256-XiWExZZbtCdl/1Koi1dTKsiLl5xSwpI8LiGxjX2yT+A=";
+                url =
+                  "https://github.com/AdguardTeam/AdGuardHome/releases/download/v0.107.32/AdGuardHome_linux_armv7.tar.gz";
+              };
+            }.${pkgs.stdenv.hostPlatform.system};
+
+          passthru.schema_version = 20;
+        });
```

I was actually thinking something in the line of:
```suggestion
    schemaVersionBefore23 = { pkgs, ... }: {
      services.adguardhome = {
        enable = true;
        settings.schema_version = 20;
```

This will cause the defaults to get populated differently. My intention was to only test that populating with an older `schema_version` style inside `settings` should still set the host/port correctly, as that changed with `schema_version` 23.

(Also note the change of name in the suggestion, this needs to be fixed in the `testScript` as well)

---

Oh, and I hope that I got the closing braces correct :sweat_smile:

---

Yeah, it makes more sense since adguard can migrate it to a higher version.